astroquery/hsa/core.py

```diff
@@ -0,0 +1,356 @@
+            If the filename is not set it will use the observation_id as filename
+            file name to be used to store the file
+        verbose : bool, optional, default False
+            flag to display information about the process
+        observation_oid : string, optional
+            Observation internal identifies. This is the database identifier
+        instrument_oid : string, optional
+            The database identifies of the instrument
+            values: 1, 2, 3
+        product_level : string, optional
+            level to download
+            values: ALL, AUXILIARY, CALIBRATION, LEVEL0, LEVEL0_5, LEVEL1, LEVEL2, LEVEL2_5, LEVEL3, ALL-LEVEL3
+
+        Returns
+        -------
+        File name of downloaded data
+        """
+        if filename is not None:
+            filename = os.path.splitext(filename)[0]
+
+        if retrieval_type is None:
+            retrieval_type = "OBSERVATION"
+
+        params = {'retrieval_type': retrieval_type}
+        if observation_id is not None:
+            params['observation_id'] = observation_id
+
+        if retrieval_type == "OBSERVATION" and instrument_name is None:
+            instrument_name = "PACS"
+
+        if instrument_name is not None:
+            params['instrument_name'] = instrument_name
+
+        link = self.data_url + "".join("&{0}={1}".format(key, val)
+                                       for key, val in params.items())
+
+        link += "".join("&{0}={1}".format(key, val)
+                        for key, val in kwargs.items())
+
+        if verbose:
+            log.info(link)
+
+        response = self._request('HEAD', link, save=False, cache=cache)
+        response.raise_for_status()
+
+        if 'Content-Type' in response.headers and 'text' not in response.headers['Content-Type']:
+            _, params = cgi.parse_header(response.headers['Content-Disposition'])
+        else:
+            error = "Data protected by propietary rights. Please check your credentials"
+            raise LoginError(error)
+
+        if filename is None:
+            if observation_id is not None:
+                filename = observation_id
+            else:
+                error = "Please set a filename for the output"
+                raise InputWarning(error)
+
+        r_filename = params["filename"]
+        suffixes = Path(r_filename).suffixes
+
+        if len(suffixes) > 1 and suffixes[len(suffixes) - 1] == ".jpg":
+            filename += suffixes[len(suffixes) - 1]
+        else:
+            filename += "".join(suffixes)
+
+        self._download_file(link, filename, head_safe=True, cache=cache)
+
+        if verbose:
+            log.info("Wrote {0} to {1}".format(link, filename))
+
+        return filename
+
+    def get_observation(self, observation_id, instrument_name, *, filename=None,
+                        verbose=False,
+                        cache=True, **kwargs):
+        """
+        Download observation from Herschel
```

Comment:
Could you say more here - what does it mean to 'download observations'? Are we downloading FITS images?  Raw timestream data?   Level X processed files? All of the above?

Reply:
I wasn't sure if this function was going to be useful or just confusing. 

You can achieve the same result using the `download_data` function with the `retrieval_type='OBSERVATION'` (which is the default value). Maybe we could delete this function completely.

Answering the question, it retrieves all the products associated with an observation (mostly FITS files).

Reply:
This might be useful, but it's necessary to explain what "all the products associated with an observation" is likely to mean.  Up to you, though.